openvdb/openvdb/tools/Merge.h

```diff
@@ -1158,30 +1161,40 @@ bool CsgDifferenceOp<TreeT>::operator()(LeafT& leaf, size_t) const
         return false;
     }
 
+    bool allequal = true;
     for (Index i = 0 ; i < LeafT::SIZE; i++) {
         const ValueT& aValue = leaf.getValue(i);
-        ValueT bValue = math::negative(mergeLeaf->getValue(i));
+        ValueT bValue = mergeLeaf->getValue(i);
+        if (PruneCancelledTiles)
```

Comment:
Probably worth doing `PruneCancelledTiles && allequal` here so the first non-match disables subsequent value checks.

Reply:
I'd sort of hope the compiler would deal with
allequal &= foo;
implying
if (allequal) allequal = foo;
for boolean foo/allequal and no side effects on foo...  We also don't get to avoid value checks because we have to load aValue and bValue in any case, so it is mostly if the cost of the branch is worth eliding a comparison and logic operation...   My random estimate is branches are more expensive, so the branch free seems best to me, but with the potential the compiler will ignore this at its discretion.

Reply:
You're probably right about that.